### PR TITLE
Choose active implementation only once.

### PR DESCRIPTION
**Pull request by vitlibar**

### CODE REVIEW

File: src/implementation.cpp [hunk 1]
```diff
@@ -58,8 +58,6 @@ class detect_best_supported_implementation_on_first_use final : public implement
 
 const detect_best_supported_implementation_on_first_use detect_best_supported_implementation_on_first_use_singleton;
 
-internal::atomic_ptr<const implementation> active_implementation{&internal::detect_best_supported_implementation_on_first_use_singleton};
```
Comment (vitlibar) [inline src/implementation.cpp:61]: There were two variables with that name: `simdjson::active_implementation` and `simdjson::internal::active_implementation`. That made the function `set_best()` choose implementation multiple times.

**Comment by lemire** [comment]
F*ck. How could we do this??????????????????????????????????????????????????????????????

**Comment by lemire** [comment]
Merging.

**Comment by lemire** [comment]
We were spending probably 100 ns on this dispatching with every call... Not enough to matter for large-ish files, but for tiny inputs, it would matter...

The cost of runtime dispatch
https://lemire.me/blog/2020/07/17/the-cost-of-runtime-dispatch/

**Pull request merged**